src/plugins/intel_npu/src/compiler_adapter/src/vcl_api.cpp

```diff
@@ -359,6 +380,85 @@ NetworkDescription VCLCompilerImpl::compile(const std::shared_ptr<const ov::Mode
+                                           updatedConfig.isAvailable(ov::intel_npu::use_base_model_serializer.name())
+                                               ? updatedConfig.get<USE_BASE_MODEL_SERIALIZER>()
+                                               : true);
+
+    std::string buildFlags;
+    const bool useIndices = !((compilerVersion.major < 5) || (compilerVersion.major == 5 && compilerVersion.minor < 9));
+
+    _logger.debug("create build flags");
+    buildFlags += driver_compiler_utils::serializeIOInfo(model, useIndices);
+    buildFlags += " ";
+    buildFlags += driver_compiler_utils::serializeConfig(config, compilerVersion);
+    _logger.debug("final build flags to compiler: %s", buildFlags.c_str());
+
+    vcl_executable_desc_t exeDesc = {serializedIR.second.get(),
+                                     serializedIR.first,
+                                     buildFlags.c_str(),
+                                     buildFlags.size()};
+    _logger.debug("compiler vcl version: %d.%d", _vclVersion.major, _vclVersion.minor);
+
+    _logger.debug("Using vclAllocatedExecutableCreateWS");
+    vcl_allocator_vector_2 allocator;
+    vcl_blob_container blocContainer;
+
+    THROW_ON_FAIL_FOR_VCL("vclAllocatedExecutableCreateWS",
+                          vclAllocatedExecutableCreateWS(_compilerHandle, exeDesc, &allocator, &blocContainer),
+                          _logHandle);
+
+    if (blocContainer.blobCount == 0 || blocContainer.blobSize == nullptr || blocContainer.blobBuffer == nullptr) {
+        OPENVINO_THROW("Failed to create VCL executable, blobCount is zero or blob is null");
+    }
+
+    // TODO fill the rest. Call "vclAllocatedExecutableCreateWS" and any other remote function required to retrieve the
+    // vector of blobs and use them to construct the vector of "NetworkDescription". The metadata objects can be empty.
+
+    std::vector<std::shared_ptr<NetworkDescription>> networkDescrs;
+    for (int i = 0; i < blocContainer.blobCount; i++) {
+        // Use empty metadata as VCL does not support metadata extraction
+        NetworkMetadata metadata;
+        networkDescrs.emplace_back(
+            std::make_shared<NetworkDescription>(std::move(*allocator.m_vector[i].second), std::move(metadata)));
+    }
+    return networkDescrs;
+}
+
+NetworkDescription VCLCompilerImpl::compileWsIterative(const std::shared_ptr<ov::Model>& model,
+                                                       const Config& config,
+                                                       size_t callNumber) const {
+    _logger.debug("compileWsIterative start");
+    const FilteredConfig* filteredConfig = dynamic_cast<const FilteredConfig*>(&config);
+    if (filteredConfig == nullptr) {
+        OPENVINO_THROW("config is not FilteredConfig");
+    }
+    FilteredConfig updatedConfig = *filteredConfig;
+    updatedConfig.update({{ov::intel_npu::ws_compile_call_number.name(), std::to_string(callNumber++)}});
```

Comment:
```suggestion
    updatedConfig.update({{ov::intel_npu::ws_compile_call_number.name(), std::to_string(callNumber)}});
```
my bad, incrementing this doesn't make sense.

Reply:
Thx! Fixed.